core/domain/feedback_services_test.py

```diff
@@ -232,7 +235,52 @@ def test_get_all_threads_without_suggestion(self):
             with self.swap(utils, 'get_time_in_millisecs',
                            self._get_milliseconds):
                 threads = feedback_services.get_all_threads(
-                    self.EXP_ID2, False)
+                    self.EXP_ID1, False)
         self.assertEqual(len(threads), 2)
-        self.assertEqual(threads[0]['thread_id'], self.THREAD_ID4)
-        self.assertEqual(threads[1]['thread_id'], self.THREAD_ID5)
+        self.assertEqual(threads[0].id, self.EXP_ID1 + '.' + self.THREAD_ID4)
+        self.assertEqual(threads[1].id, self.EXP_ID1 + '.' + self.THREAD_ID5)
+
+
+class FeedbackThreadUnitTests(test_utils.GenericTestBase):
+
+    EXP_ID = '0'
+
+    def setUp(self):
+        super(FeedbackThreadUnitTests, self).setUp()
+
+        self.viewer_id = self.get_user_id_from_email(self.VIEWER_EMAIL)
+        user_services.get_or_create_user(self.viewer_id, self.VIEWER_EMAIL)
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+
+    def test_get_all_threads(self):
+        # Create an anonymous feedback thread
+        expected_thread = {
+            'status': u'open',
+            'state_name': u'a_state_name',
+            'summary': None,
+            'original_author_username': None,
+            'subject': u'a subject'
+        }
+        feedback_services.create_thread(
+            self.EXP_ID, expected_thread['state_name'], None,
+            expected_thread['subject'], 'not used here')
+
+        threads = feedback_services.get_all_threads(self.EXP_ID, False)
+        self.assertEqual(1, len(threads))
+        self.assertDictContainsSubset(expected_thread, threads[0].to_dict())
+
+        # Viewer creates feedback thread
+        expected_thread = {
```

Comment:
expected_thread_dict (we should be careful in distinguishing thread domain objects and their dict representations)